### PR TITLE
Add test for remove command to ensure not removing according to provides

### PR DESCRIPTION
**Pull request by j-mracek**
The new behavior is more stable for cases when the same spec is used multiple times with remove command.

### CODE REVIEW

File: dnf-behave-tests/fixtures/specs/dnf-ci-provides-alternatives/bar-1.0.spec [hunk 1]
```diff
@@ -0,0 +1,23 @@
+%undefine _debuginfo_subpackages
+
+Name:           bar
+Version:        1.0
+Release:        1
+
+License:        Public Domain
```
Comment (ppisar) [inline dnf-behave-tests/fixtures/specs/dnf-ci-provides-alternatives/bar-1.0.spec:7]: Is there a reason why bar-1.0.spec URL, License and %install differ from foo.spec? Could you make them the same, so it's clear in how the two spec files differ?

**Review by ppisar — COMMENTED** [review]
Could you link the related DNF5 change <https://github.com/rpm-software-management/dnf5/pull/1126> from this commit message?

File: dnf-behave-tests/fixtures/specs/dnf-ci-provides-alternatives/bar-1.0.spec (hunk 1 above)
Comment (j-mracek) [reply dnf-behave-tests/fixtures/specs/dnf-ci-provides-alternatives/bar-1.0.spec:7]: Good point, I will unify it.

**Comment by j-mracek** [comment]
@ppisar I think I've fixed it

File: dnf-behave-tests/dnf/remove-no-provides.feature [hunk 2]
```diff
@@ -0,0 +1,27 @@
+@dnf5daemon
+Feature: Remove RPMs by provides
+
+
+Background: Install glibc
```
Comment (ppisar) [inline dnf-behave-tests/dnf/remove-no-provides.feature:5]: The Backround text is wrong. It should read "Install two packages, both providing foo".

File: dnf-behave-tests/dnf/remove-no-provides.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   Then the exit code is 0
+    And Transaction is following
+        | Action        | Package                         |
+        | install       | foo-0:1.0-1.noarch              |
+        | install       | bar-0:1.0-1.noarch              |
+
+
+
+@dnf5
+Scenario: Remove an RPM by name and nott by provide
```
Comment (ppisar) [inline dnf-behave-tests/dnf/remove-no-provides.feature:17]: "nott" is a typo. Did you mean "not"?

File: dnf-behave-tests/dnf/remove-no-provides.feature (hunk 2 above)
Comment (j-mracek) [reply dnf-behave-tests/dnf/remove-no-provides.feature]: Thank you, fixed

File: dnf-behave-tests/dnf/remove-no-provides.feature (hunk 3 above)
Comment (j-mracek) [reply dnf-behave-tests/dnf/remove-no-provides.feature]: Thank you, fixed

**Pull request merged**